otherlibs/unix/gethost.c

```diff
@@ -111,11 +111,11 @@ CAMLprim value unix_gethostbyaddr(value a)
   caml_leave_blocking_section();
   if (rc != 0) hp = NULL;
 #else
-#ifdef GETHOSTBYADDR_IS_REENTRANT
+#ifndef GETHOSTBYADDR_IS_REENTRANT
```

Comment:
This change is wrong, as well as the other changes in this file.  Explain me why the original logic is wrong and your reversed logic is right?

Reply:
It's entirely conceivable I'm wrong on this.
I would have thought that if gethostbyaddr() is reentrant, it does *not* need
to be wrapped in locking primitives, ensuring single-threaded use of gethostbyaddr()?

Reply:
You would have thought.  Please read the reference manual, http://caml.inria.fr/pub/docs/manual-ocaml/intfc.html, section 19.11.2 in particular.  Don't tinker with code you don't understand.

Reply:
OK, I stand corrected.
I could perhaps suggest to rather use the `caml_release_runtime_system` and `caml_acquire_runtime_system`
functions instead of the enter/leave function pair to make it more intuitively fathomable.